### PR TITLE
Don't use an ancestor tile if it needs to be reprojected and hasn't been

### LINKED ISSUE #4777: readyImagery is not actually ready
Reported on the forum: https://groups.google.com/forum/?hl=en#!topic/cesium-dev/Q2qkuxWVj4s

There's a crash in `GlobeSurfaceTileProvider` when you zoom into some of the tiles: 

> it happens when zoomed between tiles with X coordinates 2047, 0 and any of the poles, as well as the opposite side of the globe - between tiles 1023, 1024 and any of the poles (places where one side is loaded and other is still blurry and loading)

It only happens the first time the tiles are loaded in, not after they are cached.  I can reproduce it by opening a new incognito window and zooming in and out on the top row of numbers near the north pole.

``` javascript
var viewer = new Cesium.Viewer('cesiumContainer', {
    baseLayerPicker : true
});

var layers = viewer.imageryLayers;
var coordinateTiles = layers.addImageryProvider(Cesium.createOpenStreetMapImageryProvider({
    url : 'http://commercialplanet.eu/cesium/data6/coordinatestiles',
    maximumLevel : 11,
    credit : ''
})); 
```

```
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at addDrawCommandsForTile (http://localhost:8080/Source/Scene/GlobeSurfaceTileProvider.js:1097:27)
    at GlobeSurfaceTileProvider.endUpdate (http://localhost:8080/Source/Scene/GlobeSurfaceTileProvider.js:396:17)
    at QuadtreePrimitive.update (http://localhost:8080/Source/Scene/QuadtreePrimitive.js:306:32)
    at Globe.update (http://localhost:8080/Source/Scene/Globe.js:516:21)
    at updatePrimitives (http://localhost:8080/Source/Scene/Scene.js:2201:26)
    at executeCommandsInViewport (http://localhost:8080/Source/Scene/Scene.js:2106:9)
    at updateAndExecuteCommands (http://localhost:8080/Source/Scene/Scene.js:1974:17)
    at render (http://localhost:8080/Source/Scene/Scene.js:2404:9)
    at Scene.render (http://localhost:8080/Source/Scene/Scene.js:2442:13)
```

### PR DESCRIPTION
**Pull request by kring**
Fixes #4777 

Here's what happens to trigger the exception:
* We zoom in close to the globe and near the pole, so that we're exceeding the maximum level available from the imagery provider.
* One terrain tile, `T1`, loads an imagery tile `I1`.  `T1` doesn't cross the web mercator boundary, so reprojection of `I1` is unnecessary.  `I1` eventually enters the `READY` state without being reprojected.
* Another terrain tile, `T2`, finds that an imagery tile it would like to use is not available (missing, max level exceeded, etc.).  It walks up the parent chain from that tile until it finds `I1`.  `I1` is ready so it selects it for rendering.  But `T2` _does_ cross the web mercator boundary, so `I1` must be reprojected before it can be used.
* We try to render `T2` with `I1`, but `I1` isn't really ready for use by `T2`, so we get the "readyImagery is not actually ready" exception.

The fix is, when walking up the parent chain looking for a usable tile, to make sure that the candidate imagery tile is not only ready but that it also has been reprojected if this terrain tile needs it to be.

### CODE REVIEW

File: Specs/Scene/TileImagerySpec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*global defineSuite*/
```
Comment (pjcozzi) [inline Specs/Scene/TileImagerySpec.js]: Can you please adjust the tab spacing in this file to match the rest of Cesium?

**Comment by svetoslav80** [comment]
Seems that it's working fine now. Thanks for fixing.

**Comment by pjcozzi** [comment]
Thanks for the quick response, @svetoslav80!  It is much appreciated.

File: Specs/Scene/TileImagerySpec.js (hunk 1 above)
Comment (kring) [inline Specs/Scene/TileImagerySpec.js]: Oops, sorry, done.  I also opened #4413 which will prevent this from happening in the future.

**Comment by pjcozzi** [comment]
Thanks again @kring @svetoslav80!

**Pull request merged**